### PR TITLE
KCL-8284 - Customizable z-index for KSL highliht and edit button

### PR DESCRIPTION
**Pull request by JanBarton**
### Motivation

Which issue does this fix? Fixes #`issue number`

If no issue exists, what is the fix or new feature? Were there any reasons to fix/implement things that are not obvious?

### Checklist

- [ ] Code follows coding conventions held in this repo
- [ ] Automated tests have been added
- [ ] Tests are passing
- [ ] Docs have been updated (if applicable)
- [ ] Temporary settings (e.g. variables used during development and testing) have been reverted to defaults

### How to test

If manual testing is required, what are the steps?

### CODE REVIEW

File: src/web-components/KSLAddButtonElement.ts [hunk 1]
```diff
@@ -111,7 +111,7 @@ const templateHTML = `
     :host {
       display: inline-block;
       position: absolute;
-      z-index: 9999;
+      z-index: calc(var(--ksl-z-index, 9999) + 1);
```
Comment (kenticomartinh) [inline src/web-components/KSLAddButtonElement.ts:114]: Shouldn't this be by default also 9998, as we add +1 ?

File: src/web-components/abstract/KSLPopUpElement.ts [hunk 2]
```diff
@@ -16,7 +16,7 @@ const templateHTML = `
       display: block;
       position: absolute;
       opacity: 0;
-      z-index: 10000;
+      z-index: calc(var(--ksl-z-index, 10000) + 2);
```
Comment (kenticomartinh) [inline src/web-components/abstract/KSLPopUpElement.ts:19]: Same here

File: src/web-components/KSLHighlightElement.ts [hunk 3]
```diff
@@ -55,7 +56,7 @@ const templateHTML = `
     :host(:hover),
     :host([selected]) {
       border-color: var(--ksl-color-primary, ${Colors.Primary});
-      z-index: 20;
+      z-index: var(--ksl-z-index, 9998);
```
Comment (kenticomartinh) [inline src/web-components/KSLHighlightElement.ts:59]: This value is much higher than before, and this former 20 and below 10 are now the same, is that on purpose?

Comment (JanBarton) [reply src/web-components/KSLHighlightElement.ts:59]: I agree, I forget to keep same value ration

**Pull request merged**